Automattic/wp-calypso
# Show Finish Setup warnings for Google Workspace on Email Plan page

#### Summary

As a follow up to #52924, this PR adds logic to optionally display a warning and a `Finish Setup` CTA prompt to users when viewing a Google Workspace  email plan when the TOS has not been accepted at the account/domain level.

Additionally: 
- Some methods that operate on email accounts are moved centrally to the `client/lib/emails` folder to make it easier to access them for reuse.
- We are now opting to display just the first item in the warnings array so as to improve UX and optimise screen real estate.

#### Testing instructions




- Checkout this branch in your sandbox, or try it out on [calypso live](https://calypso.live/?branch=add/add-google-tos-warnings-to-email-plan)
- Navigate to a site i.e. `/email/:siteName:` where you have just bought a Google Workspace plan afresh and haven't logged in to accept the TOS yet
- Confirm that status of the plan is `Action Required`
- Confirm that warnings are shown just below the status  and a CTA to `Finish Setup`
- Confirm that the button leads you to accept the TOS


<img width="1064" alt="Screenshot 2021-05-21 at 6 05 18 AM" src="https://user-images.githubusercontent.com/277661/119084959-a5d38f00-b9fa-11eb-9b53-6b146efbfaae.png">
